linera-views/src/backends/dynamo_db.rs

```diff
@@ -343,32 +353,23 @@ impl AdminKeyValueStore for DynamoDbStoreInternal {
             .map(|n| Arc::new(Semaphore::new(n)));
         let max_stream_queries = config.max_stream_queries;
         let namespace = namespace.to_string();
-        let start_key = extend_root_key(&[]);
         let store = Self {
             client,
             namespace,
             semaphore,
             max_stream_queries,
-            start_key,
-            root_key_written: Arc::new(AtomicBool::new(false)),
         };
         Ok(store)
     }
 
-    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self, DynamoDbStoreInternalError> {
-        let client = self.client.clone();
-        let namespace = self.namespace.clone();
-        let semaphore = self.semaphore.clone();
-        let max_stream_queries = self.max_stream_queries;
-        let start_key = extend_root_key(root_key);
-        Ok(Self {
-            client,
-            namespace,
-            semaphore,
-            max_stream_queries,
-            start_key,
-            root_key_written: Arc::new(AtomicBool::new(false)),
-        })
+    fn open_exclusive(&self, root_key: &[u8]) -> Result<Self::Store, DynamoDbStoreInternalError> {
+        let mut start_key = EMPTY_ROOT_KEY.to_vec();
+        start_key.extend(root_key);
+        self.open_internal(start_key)
+    }
+
+    fn open_shared(&self, root_key: &[u8]) -> Result<Self::Store, DynamoDbStoreInternalError> {
+        self.open_exclusive(root_key)
```

Comment:
If the opening of a shared one is the same as opening an exclusive one, then we should have an `open_store` and then both `open_shared/open_exclusive` that refer to it.

Reply:
see other comment